Apps/W1/Shopify/app/src/Payments/Codeunits/ShpfyPayments.Codeunit.al

```diff
@@ -203,4 +203,48 @@ codeunit 30169 "Shpfy Payments"
             end;
         end;
     end;
+
+    internal procedure UpdateDisputeStatus()
+    var
+        PaymentTransaction: Record "Shpfy Payment Transaction";
+        OrderId: BigInteger;
+        JDisputes: JsonArray;
+        JItem: JsonToken;
+        JResponse: JsonToken;
+        Url: Text;
+        UrlTxt: Label 'shopify_payments/disputes.json?limit=250', Locked = true;
```

Comment:
It needs to go through all disputes, because disputes get updated. The id won't change, but the status and finalized_on will change over time.
If we limit it to new disputes only, we will only fetch the data in the moment of the synchronization.

Reply:
Sorry but still not convinced on this, looks like disputes will grow over time. [Shopify docs](https://shopify.dev/docs/api/admin-rest/2023-10/resources/dispute) do not mention anything about deleting them, so there will be at a point in time where we will go through many disputes for no reason. That would be a painful performance headache.

Maybe this should be the other way around: 
- Filter payment transactions on non won/lost dispute status 
- Loop through payment transactions and get disputes for each
- Update them if necessary

Reply:
@onbuyuka 

I apologize for the delay, it was a super busy week. Updated the code to flip the update logic, and adapted the tests accordingly. Removed next page loops considering we're pulling one record at a time.